### PR TITLE
chore: Enable semantic pr check in rust-libp2p

### PR DESCRIPTION
**Pull request by galargh**
### Summary

This is a follow up to #93 and a replacement for prematurely merged https://github.com/libp2p/github-mgmt/pull/121.

### What else do we need to know?

**DRI:** @p-shahi 

### Reviewer's Checklist

- [ ] It is clear where the request is coming from (if unsure, ask)
- [ ] All the automated checks passed
- [ ] The YAML changes reflect the summary of the request
- [ ] The Terraform plan posted as a comment reflects the summary of the request

### CODE REVIEW

**Review by thomaseizinger — APPROVED** [review]

**Comment by thomaseizinger** [comment]
Do we also want to include a validation of the PR-title length? We have the following:

```yml
      - name: Check PR title length
        env:
          TITLE: ${{ github.event.pull_request.title }}
        run: |
          title_length=${#TITLE}
          if [ $title_length -gt 72 ]
          then
            echo "PR title is too long (greater than 72 characters)"
            exit 1
          fi
```

**Comment by p-shahi** [comment]
That's a good idea @thomaseizinger feel free to push changes directly

**Comment by thomaseizinger** [comment]
> That's a good idea @thomaseizinger feel free to push changes directly

I don't have push rights to this repo so opened a PR: https://github.com/libp2p/github-mgmt/pull/129

Also, I want to note that when squash merging PRs, GitHub likes to add the pull request number. This is **not** accounted for in this limit. In `rust-libp2p`, we merge using mergify which creates the commit message according to the following template:

https://github.com/libp2p/rust-libp2p/blob/30ff1317d6850422eb2d8049846f5d7f34eb3575/.github/mergify.yml#L6-L10

**Comment by galargh** [comment]
> Do we also want to include a validation of the PR-title length? We have the following:
> 
> ```yaml
>       - name: Check PR title length
>         env:
>           TITLE: ${{ github.event.pull_request.title }}
>         run: |
>           title_length=${#TITLE}
>           if [ $title_length -gt 72 ]
>           then
>             echo "PR title is too long (greater than 72 characters)"
>             exit 1
>           fi
> ```

Why 72 characters? Personally, I'd be really surprised to see my PR failing because of title length restrictions. I think it might be worth pulling in @achingbrain again because that's something that's gonna' affect all the JS repos here now.

**Comment by thomaseizinger** [comment]
It is a Git convention. Subject line should be at most 72 and body wrapped at 80.

GitHub will wrap your subject line if it is longer than 72 chars which looks a bit ugly when you read the commit message in the web UI.

Other tools (like `git` itself) make similar assumptions about the structure of the commit message.

**Comment by thomaseizinger** [comment]
To be clear, I don't think we need to enforce this ORG-wide. It is a bit of a nitpick. I only suggested it because it is what we do in `rust-libp2p` and thought it might be interesting to discuss. I am more than happy to just enforce this in our repo separately if it causes to much friction!

Let me know what you think :)

File: github/libp2p.yml
Comment (marten-seemann) [inline github/libp2p.yml:2071]:
```suggestion
```

**Review by marten-seemann — DISMISSED** [review]

File: github/libp2p.yml
Comment (MarcoPolo) [reply github/libp2p.yml:2072]: Why not? It would be nice to follow this so we can auto-highlight breaking changes.

Comment (marten-seemann) [reply github/libp2p.yml:2072]: It seems like a major annoyance (especially the PR title character limit).
We had similar discussions when we considered which linters to use in Unified CI. Yes, more linters can be helpful, but if you enable too many of them, they'll get in the way of productive work. That's why we only have `go vet` and `staticcheck`.

Comment (thomaseizinger) [reply github/libp2p.yml:2072]: > Why not? It would be nice to follow this so we can auto-highlight breaking changes.

FWIW, we are unlikely to use the "highlight" breaking changes bit in any form of automated tooling because a PR can easily touch multiple modules and it is thus not clear, which one this refers to once the PR is squashed.

That being said, it is still nice to highlight.

Comment (marten-seemann) [reply github/libp2p.yml:2072]: For example, this one would fail: https://github.com/libp2p/go-libp2p-backoff/pull/11

Comment (thomaseizinger) [reply github/libp2p.yml:2072]: Yep, we have had an occasional dependabot PR with too long titles. In those cases, I've just shortened it.

Comment (p-shahi) [reply github/libp2p.yml]: In the go-libp2p meeting today, we decided to not enable this in go-libp2p for now.

**Review by p-shahi — APPROVED** [review]
needs @thomaseizinger or @mxinden approval before merging

**Review by thomaseizinger — APPROVED** [review]
Good to go in from my end!

**Comment by thomaseizinger** [comment]
Is this going to by synced in via a PR or how is this going to land in our repository?

**Comment by p-shahi** [comment]
@thomaseizinger it should apply via Terraform but the job failed: https://github.com/libp2p/github-mgmt/actions/runs/4287574003/jobs/7468603098 @galargh

**Comment by thomaseizinger** [comment]
Our repository uses branch protection though. How can terraform add a file here without going through a PR? Does it have a backdoor to just modify our Git repository?

**Pull request merged**
